packages/aws-cdk/lib/api/cloudwatch-logs.ts

```diff
@@ -0,0 +1,94 @@
+import * as cxapi from '@aws-cdk/cx-api';
+import { Mode, SdkProvider } from './aws-auth';
+import { EvaluateCloudFormationTemplate, LazyListStackResources } from './evaluate-cloudformation-template';
+import { CloudWatchLogEventMonitor } from './monitor/logs-monitor';
+
+// resource types that have associated CloudWatch Log Groups that should
+// _not_ be monitored
+const excludedTypes: string[] = ['AWS::EC2::FlowLog', 'AWS::CloudTrail::Trail'];
+
+// resource types that will create a cloudwatch log group
+// with a specific name if one is not provided
+const implicitTypes: string[] = ['AWS::Lambda::Function', 'AWS::CodeBuild::Project'];
+
+export async function registerCloudWatchLogGroups(
+  sdkProvider: SdkProvider,
+  stackArtifact: cxapi.CloudFormationStackArtifact,
+  cloudWatchLogMonitor: CloudWatchLogEventMonitor,
+): Promise<void> {
+
+  const logGroupNames = new Array<string>();
+  const logGroupLogicalIds = new Set<string>();
+  const implicitMap = new Map<string, string>();
+  //
+  // resolve the environment, so we can substitute things like AWS::Region in CFN expressions
+  const resolvedEnv = await sdkProvider.resolveEnvironment(stackArtifact.environment);
+  // create a new SDK using the CLI credentials, because the default one will not work for new-style synthesis -
+  // it assumes the bootstrap deploy Role, which doesn't have permissions to update Lambda functions
```

Comment:
Hmm. Is this comment still true, given that we're not updating Lambda functions here?